examples/brute_force/brute_force.cpp

```diff
@@ -0,0 +1,166 @@
+  }
+};
+
+struct PrintfCallback
+{
+  Kokkos::View<int, ExecutionSpace, Kokkos::MemoryTraits<Kokkos::Atomic>> c_;
+  PrintfCallback()
+      : c_{"counter"}
+  {
+  }
+  template <typename Predicate, typename OutputFunctor>
+  KOKKOS_FUNCTION void operator()(Predicate const &predicate, int i,
+                                  OutputFunctor const &out) const
+  {
+    int const j = getData(predicate);
+    printf("%d callback (%d,%d)\n", ++c_(), i, j);
+    out(i);
+  }
+};
+
+template <typename T, typename... P>
+std::vector<T> view2vec(Kokkos::View<T *, P...> view)
+{
+  std::vector<T> vec(view.size());
+  Kokkos::deep_copy(Kokkos::View<T *, Kokkos::HostSpace,
+                                 Kokkos::MemoryTraits<Kokkos::Unmanaged>>(
+                        vec.data(), vec.size()),
+                    view);
+  return vec;
+}
+
+template <typename OutputView, typename OffsetView>
+void print(OutputView const out, OffsetView const offset)
+{
+  int const n_queries = offset.extent(0) - 1;
+
+  auto const h_out = view2vec(out);
+  auto const h_offset = view2vec(offset);
+  int count = 0;
+
+  for (int j = 0; j < n_queries; ++j)
+    for (int k = h_offset[j]; k < h_offset[j + 1]; ++k)
+      printf("%d result (%d, %d)\n", ++count, h_out[k], j);
+}
+
+int main(int argc, char *argv[])
+{
+  Kokkos::ScopeGuard guard(argc, argv);
+
+  int nqueries = 5, nprimitives = 5, nrepeats = 1;
+  namespace bpo = boost::program_options;
+  bpo::options_description desc("Allowed options");
+  // clang-format off
+  desc.add_options()
+      ( "help", "help message" )
+      ( "predicates", bpo::value<int>(&nqueries), "number of predicates" )
+      ( "primitives", bpo::value<int>(&nprimitives), "number of primitives" )
+      ( "iterations", bpo::value<int>(&nrepeats), "number of iterations" )
+      ;
+  // clang-format on
+  bpo::variables_map vm;
+  bpo::store(bpo::command_line_parser(argc, argv).options(desc).run(), vm);
+  bpo::notify(vm);
+
+  if (vm.count("help") > 0)
+  {
+    std::cout << desc << '\n';
+    return 1;
+  }
+  printf("Primitives: %d\n", nprimitives);
+  printf("Predicates: %d\n", nqueries);
+  printf("Iterations: %d\n", nrepeats);
+
+  ExecutionSpace space{};
+  Dummy primitives{nprimitives};
+  Dummy predicates{nqueries};
+
+  for (int i = 0; i < nrepeats; i++)
+  {
+    int out_count;
+    {
+      Kokkos::Timer timer;
+      ArborX::BoundingVolumeHierarchy<MemorySpace> bvh{space, primitives};
+
+      Kokkos::View<int *, ExecutionSpace> indices("indices_ref", 0);
+      Kokkos::View<int *, ExecutionSpace> offset("offset_ref", 0);
+      bvh.query(
+          space, predicates, ArborX::Details::DefaultCallback{}, indices,
+          offset,
+          ArborX::Experimental::TraversalPolicy{}.setPredicateSorting(true));
+
+      double time = timer.seconds();
+      if (i == 0)
+        printf("Collisions: %.5f\n",
+               (float)(indices.extent(0)) / (nprimitives * nqueries));
```

Comment:
This is an interesting metric. I am struggling a bit to understand its meaning. For example, if my primitives are points uniformly at random distributed with a constant density, and my predicates sphere with constant radius around those points, then the number of intersections per predicate is constant, lets say C. However, this metric is then approximately `C / nprimitives` and goes to zero with increasing number.

Reply:
The point of the metric is to give an idea of how many collisions you find compared to the size of the search space. 
In your example the number of collision is constant but there are more combinations of predicates to primitives that need to be checked in order to find them so the amount of useful work done by the brute force decreases with more primitives/predicates. I am fine with any other metric, I think it's one way of understanding when brute force might be useful, but there are many other ways.